compiler/packages/babel-plugin-react-compiler/src/Entrypoint/Program.ts

```diff
@@ -276,41 +277,27 @@ export function compileProgram(
   program: NodePath<t.Program>,
   pass: CompilerPass,
 ): void {
-  if (pass.opts.sources) {
-    if (pass.filename === null) {
-      const error = new CompilerError();
-      error.pushErrorDetail(
-        new CompilerErrorDetail({
-          reason: `Expected a filename but found none.`,
-          description:
-            "When the 'sources' config options is specified, the React compiler will only compile files with a name",
-          severity: ErrorSeverity.InvalidConfig,
-          loc: null,
-        }),
-      );
-      handleError(error, pass, null);
-      return;
-    }
-
-    if (!isFilePartOfSources(pass.opts.sources, pass.filename)) {
-      return;
-    }
-  }
-
-  // Top level "use no forget", skip this file entirely
-  if (
-    findDirectiveDisablingMemoization(program.node.directives, pass.opts) !=
-    null
-  ) {
+  if (shouldSkipCompilation(program, pass)) {
     return;
   }
 
-  const environment = parseEnvironmentConfig(pass.opts.environment ?? {});
+  /*
+   * TODO(lauren): Remove pass.opts.environment nullcheck once PluginOptions
```

Comment:
Previously, we waited until compiling a function to validate environment configs. Let's just do this unconditionally -- this should be a fatal error and hopefully never happens in prod